webapp/store/models.py

```diff
@@ -0,0 +1,172 @@
+        'archive_url': cs.archive_url(ref),
+        'bugs_url': bugs_url,
+        'bzr_url': bzr_url,
+        'entity_data': entity_data,
+        'files': _getEntityFiles(ref, meta['manifest']),
+        'homepage': homepage,
+        'icon': _parse_charm_icon(charm_or_bundle_id),
+        'id': charm_or_bundle_id,
+        'latest_revision': _expand_revision(revision_list[0], meta['charm-metadata']['Name']),
+        'options': meta.get('charm-config', {}).get('Options'),
+        'owner': entity_data.get('owner', {}).get('User'),
+        'provides': meta['charm-metadata'].get('Provides'),
+        'requires': meta['charm-metadata'].get('Requires'),
+        'resources': _extract_resources(ref, meta.get('resources')),
+        'readme': _render_markdown(cs.entity_readme_content(charm_or_bundle_id)),
+        'revision_list': revision_list,
+        'revision_number': ref.revision,
+        'revisions': revisions,
+        'series': meta.get('supported-series', {}).get('SupportedSeries'),
+        'is_charm': 'charm-metadata' in meta
+    }
+
+    return charm_or_bundle
+
+
+def _getEntityFiles(ref, manifest=None):
+    try:
+        files = cs.files(ref, manifest=manifest)
+    except EntityNotFound:
+        files = {}
+    if not files:
+        files = {}
+    try:
+        readme = cs.entity_readme_content(ref)
+    except EntityNotFound:
+        readme = None
+    return {
+        'files': files,
+        'readme': readme,
+    }
+
+
+def _parse_charm_icon(charm_id):
+    try:
+        return cs.charm_icon_url(charm_id)
+    except Exception:
+        return abort(500, "Entity not found {}".format(charm_id))
+
+
+def _expand_revision(revision_id, charm_name):
+    '''From a revision id, create a Dict.
+    @param revision_id A revision id string.
+    @param charm_name The name of the charm
+    '''
+
+    id = revision_id.split('-')[-1]
+    url = charm_name + '/' + id
+    return {
+        'id': int(id),
+        'full_id': revision_id,
+        'url': url}
+
+
+def _extract_resources(ref, data):
+    """Extract data from resources metadata.
+    @param ref The reference of the entity.
+    @param data the resources metadata associated with the entity.
+    @return a dictionary of resource name and an array containing
+            the file extension, the file link of the resource.
+    """
+    if not data:
+        # if resources are not present return an empty dictionary
+        return {}
+
+    try:
+        resources = {}
+        for i in data:
+            resource_url = ""
+            if i['Revision'] >= 0:
+                resource_url = self.cs.resource_url(ref,
+                                                    i['Name'],
+                                                    i['Revision'])
+            resources[i['Name']] = [os.path.splitext(i['Path'])[1],
+                                    resource_url]
+    except (KeyError, TypeError, ValueError):
+        log.info('Unable to read resources for ',
```

Comment:
I don't think `log` is defined anywhere, so will error if you ever hit this block.

Reply:
I'm also not super happy about this except block either - it seems to be fairly fuzzily catching a whole host of errors which suggest we don't understand our data model terribly well. Surely if `Name` or `Revision` are missing, or of the wrong type, we should know why, and be avoiding requesting them in those cases?

Maybe try without the `try..except` block until we actually see the error and then we can get to the bottom of when it's happening?